docs/src/usage.md

```diff
@@ -1,3 +1,74 @@
 # Usage
 
-TODO
+If you've not read the [Julia Base Broadcast Background](@ref) section, that may
+be helpful in understanding concepts, types, and functions discussed here.
+
+The basic usage of this package involves wrapping intermediate broadcast
+expressions with dot-calls to `lazy_broadcast`, and then working with those
+resulting objects. Let's make one now.
+
+```@example usage
+using LazyBroadcast;
+x = rand(10);
+y_lazy = lazy_broadcast.(x .+ x);
+```
+
+As you can see, `y_lazy` is a `Broadcasted` object. There are several things
+that you can do with them:
+ 
+ - `materialize` them (which executes the expression)
+ - Combine with other broadcast expressions
+ - Call functions that support `Broadcasted` objects
+
+Let's try each of these.
+
+## Materialize
+
+This is as simple as calling `Base.Broadcast.materialize`
+
+```@example usage
+y_array = Base.Broadcast.materialize(y_lazy)
+```
+Or, we can call the in-place version, `Base.Broadcast.materialize!`:
+
+```@example usage
+z_array = similar(x)
+Base.Broadcast.materialize!(z_array, y_lazy)
+z_array
+```
+
+## Combine with other broadcast expressions
+
+This is as simple as combining expressions with more dot operations:
+
+```@example usage
+z_lazy = lazy_broadcast.(x .* x);
+fused_lazy = y_lazy .+ z_lazy # Equivalent to `x .+ x .+ x .* x`
+```
+
+In this combined expression, `y_lazy .+ z_lazy` is equivalent to `x .+ x .+ x .*
+x`, which means that this entire expression is fused. If we want to further
+delay execution, we can wrap the entire result in a dot-`lazy_broadcast` call:
+
+```@example usage
+so_lazy = lazy_broadcast.(y_lazy .+ z_lazy)
+```
+Now, we have a `Broadcasted` object again.
+
+## Call functions that support 
+
+Several functions in Julia Base support operations on `Broadcasted` objects.
+For example, `sum`:
+
+```@example usage
+sum(so_lazy)
+```
+
+Another common one is `Base.copyto!`, which can be called naturally through
+broadcast expressions (`@.` -> `materialize!` -> `copyto!`, see
+[Julia Base Broadcast Background](@ref) for details):
+
```

Comment:
Are there more examples of functions like these ones?

How does a user know which function behave like this with broadcasted?

Reply:
No, I don't think so, at least not in general-- it's an implementation detail, and the method table is global, so anyone can write methods on broadcasted objects.

I could dig around for methods in Base, but I don't think it's practical to try and document a complete list. Another one that I see that is relevant is `Base.similar`.

But maybe more importantly, `Broadcasted` objects typically behave very much like arrays, which I think is documented in the official Julia documentation.

Reply:
I'm going to merge this, but maybe we can discuss this further in a live review?